code/game/objects/items/devices/scanners.dm

```diff
@@ -28,9 +28,14 @@ MASS SPECTROMETER
 		START_PROCESSING(SSobj, src)
 
 /obj/item/device/t_scanner/proc/flick_sonar(obj/pipe)
-	var/image/I = image('icons/effects/effects.dmi', pipe, "blip", pipe.layer+1)

```

Comment:
It should not work on the floor, as the whole point of the refactor is to only show the image to the user.
As for the backpack i'd say that it's reasonable to not make it work, much like you have to hold flashlights on your pockets to emit light, you have to hold the scanner in your pockets/belt/hands to emit t-rays.